host/xtest/xtest_20000.c

```diff
@@ -459,13 +460,15 @@ static void storage_corrupt(ADBG_Case_t *c,
 		Do_ADBG_BeginSubCase(c, "| filename: %s , data size: %d byte(s)",
 				     filename, tv->data_len);
 
-		ADBG_EXPECT(c, TEE_SUCCESS,
+		if (ADBG_EXPECT(c, TEE_SUCCESS,
 			    obj_create(&sess, filename, ARRAY_SIZE(filename),
 				TEE_DATA_FLAG_ACCESS_WRITE, 0, filedata,
-				tv->data_len, &obj_id));
+			        tv->data_len, &obj_id)))
 
-		ADBG_EXPECT(c, TEE_SUCCESS,
-			    obj_close(&sess, obj_id));
+			ADBG_EXPECT(c, TEE_SUCCESS,
+			            obj_close(&sess, obj_id));
+		else
+			goto exit;
```

Comment:
Possible memleak. By exiting here we don't free filedata (allocation is on line 451).
The same issue on lines 503/533